subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependency.java

```diff
@@ -214,13 +217,26 @@ public interface ModuleDependency extends Dependency, HasConfigurableAttributes<
     ModuleDependency capabilities(Action<? super ModuleDependencyCapabilitiesHandler> configureAction);
 
     /**
-     * Returns the set of requested capabilities for this dependency.
-     * @return An immutable view of requested capabilities. Updates must be done calling {@link #capabilities(Action)}.
+     * Returns the set of explicitly requested capabilities for this dependency.
+     * <p>
+     * Prefer {@link #getCapabilitySelectors()}.
+     *
+     * @return An immutable view of all explicitly requested capabilities. Updates must be done calling {@link #capabilities(Action)}.
      *
      * @since 5.3
      */
     List<Capability> getRequestedCapabilities();
 
+    /**
+     * Returns the set of capabilities that are requested for this dependency
+     *
+     * @return A view of all requested capabilities. Updates must be done calling {@link #capabilities(Action)}.
+     *
+     * @since 8.11
+     */
+    @Incubating
+    Provider<Set<CapabilitySelector>> getCapabilitySelectors();
```

Comment:
Not sure how I feel about this being a provider

Reply:
I would be skeptical of use cases where someone needs to "wire off" of the capability selectors of a given dependency

Reply:
We changed this to not return a provider, since users should not be wiring off this property